Makefile

```diff
@@ -0,0 +1,363 @@
+
+# https://github.com/knative/eventing/releases
+KNATIVE_EVENTING_VERSION ?= 1.0.0
+install-knative-eventing: | $(KUBECONFIG) $(KUBECTL) ## Install Knative Eventing
+	$(KUBECTL) $(K_CMD) --filename \
+		https://github.com/knative/eventing/releases/download/knative-v$(KNATIVE_EVENTING_VERSION)/eventing-crds.yaml
+	$(KUBECTL) $(K_CMD) --filename \
+		https://github.com/knative/eventing/releases/download/knative-v$(KNATIVE_EVENTING_VERSION)/eventing-core.yaml
+	$(KUBECTL) wait --for=condition=available deploy/eventing-controller --timeout=30s --namespace $(SYSTEM_NAMESPACE)
+	$(KUBECTL) wait --for=condition=available deploy/eventing-webhook --timeout=30s --namespace $(SYSTEM_NAMESPACE)
+
+install-knative-eventing-rabbitmq: | $(KUBECONFIG) $(KO) install-knative-eventing ## Install dev Knative Eventing RabbitMQ - also installs Knative Eventing
+	$(KO) apply --filename config/broker
+	$(KUBECTL) wait --for=condition=available deploy/rabbitmq-broker-controller --timeout=30s --namespace $(SYSTEM_NAMESPACE)
+	$(KUBECTL) wait --for=condition=available deploy/rabbitmq-broker-webhook --timeout=30s --namespace $(SYSTEM_NAMESPACE)
+	$(KO) apply --filename config/source
+	$(KUBECTL) wait --for=condition=available deploy/pingsource-mt-adapter --timeout=30s --namespace knative-eventing
+
+.PHONY: e2e-setup
+e2e-setup: install-rabbitmq-cluster-operator install-rabbitmq-topology-operator install-knative-eventing-rabbitmq ## Setup environment for e2e testing
+	@printf "$(INFO)Starting point: $(BOLD).github/workflows/kind-e2e.yaml$(RESET)\n"
+	@printf "$(INFO)This is slow - $(BOLD)33s cached$(RESET)$(INFO) - so it is kept as a non-explicit dependency for any e2e test$(RESET)\n"
+
+.PHONY: e2e-setup-rm
+e2e-setup-rm: | $(KIND)
+	$(KIND) delete cluster --name $(KIND_CLUSTER_NAME)
+
+.PHONY: test-e2e-publish
+test-e2e-publish: ## Run TestKoPublish end-to-end tests
+	@printf "$(WARN)Remember to run $(BOLD)make e2e-setup$(RESET)$(WARN) if needed$(RESET)\n"
+	go test -v -race -count=1 -timeout=15m -tags=e2e ./test/e2e/... -run 'TestKoPublish' \
+	| grep -v "no test files"
+
+.PHONY: test-e2e-broker
+test-e2e-broker: ## Run Broker end-to-end tests
+	@printf "$(WARN)Remember to run $(BOLD)make e2e-setup$(RESET)$(WARN) if needed$(RESET)\n"
+	@printf "$(WARN)$(BOLD)rabbitmqcluster$(RESET)$(WARN) has large resource requirements 🤔$(RESET)\n"
+	go test -v -race -count=1 -timeout=15m -tags=e2e ./test/e2e/... -run 'Test.*Broker.*' \
+	| grep -v "no test files"
+
+.PHONY: test-e2e-source
+test-e2e-source: ## Run Source end-to-end tests
+	@printf "$(WARN)Remember to run $(BOLD)make e2e-setup$(RESET)$(WARN) if needed$(RESET)\n"
+	go test -v -race -count=1 -timeout=15m -tags=e2e ./test/e2e/... -run 'Test.*Source.*' \
+	| grep -v "no test files"
+
+.PHONY: test-e2e
+test-e2e: test-e2e-publish test-e2e-broker test-e2e-source
+
+BUILD_TAGS=e2e
+.PHONY: build
+build: ## Build binaries with e2e tags
```

Comment:
I wouldn't call this `build`, since it doesn't actually build any binaries. It's more a test that things *can* build, the compiler makes no warnings. We use `go test` to test that tests also compile, regular `go build` doesn't build tests apparently.

Reply:
This is the `Build` step from this GitHub Action workflow: https://github.com/knative-sandbox/eventing-rabbitmq/blob/23a39265b91a2e9f1b4729f12a6f6f94a5ac906f/.github/workflows/knative-go-build.yaml#L51-L65

What do you think that we should call it instead?

Reply:
Maybe test-compilation?

Reply:
Fixed in https://github.com/knative-sandbox/eventing-rabbitmq/pull/525/commits/e5e6bbf2d50b0e2c8f432022e3429cb2e187091a